src/components/application_manager/include/application_manager/command_factory.h

```diff
@@ -54,6 +58,98 @@ class CommandFactory {
   virtual CommandSharedPtr CreateCommand(
       const commands::MessageSharedPtr& message,
       commands::Command::CommandSource source) = 0;
+  virtual bool IsAbleToProcess(
+      const int32_t,
+      const application_manager::commands::Command::CommandSource) const = 0;
+};
+
+class CommandCreator {
+ public:
+  virtual ~CommandCreator() {}
+  virtual bool CanBeCreated() const = 0;
+  virtual CommandSharedPtr create(
+      const commands::MessageSharedPtr& message) const = 0;
+};
+
+template <typename CommandType>
+class DefaultCommandCreator : public CommandCreator {
+ public:
+  DefaultCommandCreator(ApplicationManager& application_manager,
+                        RPCService& rpc_service,
+                        HMICapabilities& hmi_capabilities,
+                        PolicyHandlerInterface& policy_handler)
+      : application_manager_(application_manager)
+      , rpc_service_(rpc_service)
+      , hmi_capabilities_(hmi_capabilities)
+      , policy_handler_(policy_handler) {}
+
+ private:
+  bool CanBeCreated() const override {
+    return true;
+  }
+
+  CommandSharedPtr create(
+      const commands::MessageSharedPtr& message) const override {
+    CommandSharedPtr command(new CommandType(message,
+                                             application_manager_,
+                                             rpc_service_,
+                                             hmi_capabilities_,
+                                             policy_handler_));
+    return command;
+  }
+
+  ApplicationManager& application_manager_;
+  RPCService& rpc_service_;
+  HMICapabilities& hmi_capabilities_;
+  PolicyHandlerInterface& policy_handler_;
+};
+
+struct InvalidCommand {};
+
+template <>
+class DefaultCommandCreator<InvalidCommand> : public CommandCreator {
+ public:
+  DefaultCommandCreator(ApplicationManager& application_manager,
+                        RPCService& rpc_service,
+                        HMICapabilities& hmi_capabilities,
+                        PolicyHandlerInterface& policy_handler) {
+    UNUSED(application_manager);
+    UNUSED(rpc_service);
+    UNUSED(hmi_capabilities);
+    UNUSED(policy_handler);
+  }
+
+ private:
+  bool CanBeCreated() const override {
+    return false;
+  }
+  CommandSharedPtr create(
+      const commands::MessageSharedPtr& message) const override {
+    UNUSED(message);
+    return CommandSharedPtr();
+  }
+};
+
+struct CommandCreatorFacotry {
```

Comment:
@BSolonenko CommandCreatorFacotry -> CommandCreatorFactory

Reply:
@LuxoftAKutsan  fixed in 75224d1

Reply:
@LitvinenkoIra   fixed in 75224d1